services/app-api/handlers/banners/fetch.ts

```diff
@@ -2,13 +2,17 @@ import handler from "../handler-lib";
 // utils
 import { error } from "../../utils/constants/constants";
 import { getBanner } from "../../storage/banners";
-import { badRequest, ok } from "../../utils/responses/response-lib";
+import { badRequest, noContent, ok } from "../../utils/responses/response-lib";
 
 export const fetchBanner = handler(async (event, _context) => {
-  if (!event?.pathParameters?.bannerId!) {
+  const bannerId = event?.pathParameters?.bannerId;
+  if (!bannerId) {
     return badRequest(error.NO_KEY);
   }
-  const bannerId = event?.pathParameters?.bannerId!;
   const banner = await getBanner(bannerId);
-  return ok(banner);
+  if (banner) {
+    return ok(banner);
+  }
+  // Instead of 404, return 204 for silent error
+  return noContent();
```

Comment:
discussion: i believe the reason we originally used a 404 is because the banner fetch is fetching on a specific id. granted, it's the id we use for all banners. i think we've flipped between 204 and 404 before. i am fine with 204, just calling it out in case anyone wants to get api nerdy with me 🤓

Reply:
Right now, the banner doesn't 404. It gives a 200 even if it doesn't exist (uses undefined body). So, in my comment, I meant that a 404 is the more logical choice here, but it seems 204 is more in line with what we were doing with the 200.